internal/outgoing/dispatcher.go

```diff
@@ -48,22 +51,37 @@ func (d *Dispatcher) Dispatch(
 			d.Errors = append(d.Errors, err)
 
 			// Send without throttling
-			go d.sendEvent(e, destination)
+			go d.sendEvent(e, destination, documentStore)
 			continue
 		}
 
 		if canTrigger {
-			go d.sendEvent(e, destination)
+			go d.sendEvent(e, destination, documentStore)
 		} else {
-			d.processDelayedEvent(e, timeRemain, destination, store)
+			d.processDelayedEvent(e, timeRemain, destination, store, documentStore)
 		}
 	}
 	return nil
 }
 
 // Private Functions
+func (d *Dispatcher) cloneEventWithDocument(e models.IncomingEvent, destination models.Destination, documentStore interfaces.DocumentStoreInterface) models.IncomingEvent {
+	if destination.Config.IncludeDocument == false {
+		return e;
+	}
+
+	// memoize document to be used across events for diff. destinations
+	if d.targetDocument == nil {
+		d.targetDocument = documentStore.GetDocument(e)
+	}
+
+	clone := e
```

Comment:
this will create a new IncomingEvent instance with all the attributes referencing the old instance, is it expected lol?

Reply:
or we just merge the document attributes into the event payload?

Reply:
It was intended to avoid unexpectedly injecting document into events with destinations that does not need the document

If we merge the document into the payload, similar handling is probably still needed (while beanstalkd sender will now be able include the document)

Reply:
oh didn't think of that lol maybe we can convert the payload into a JSON string and pass that as an argument

Reply:
Then it would be difficult to send the whole event as a json string (e.g, HTTP Sender) where I'll have to unmarshal the payload and then inject the payload into the event and marshal it again